ambry-messageformat/src/main/java/com.github.ambry.messageformat/PutMessageFormatInputStream.java

```diff
@@ -29,12 +29,16 @@
  *  - - - - - - - - - - - - - -
  * |       blob key            |
  *  - - - - - - - - - - - - - -
+ * |     Encryption key        |
+ *  - - - - - - - - - - - - - -
  * |  Blob Properties Record   |
  *  - - - - - - - - - - - - - -
  * |  User metadata Record     |
  *  - - - - - - - - - - - - - -
  * |       Blob Record         |
  *  - - - - - - - - - - - - - -
+ * |            CRC            |
```

Comment:
You should omit this. We have CRCs for individual subrecords, not for the whole record (which makes it more flexible - you only have to read a subrecord to be able to verify CRC). The `CRC` you see being computed below is only for the Blob Record. For the rest of the subrecords, the serialize method itself does it.

Reply:
Thanks. Addressed.